### PR TITLE
Add support for composable forge tests

### PR DESCRIPTION
**Pull request by igor-aptos**
Simplify the testing code, by allowing any two tests to be composed, instead of needing to have tests with "inner" tests defined.

Additionally, PerformanceTestWithFN is identical to PerformanceTest, both send to FN if there are any, and to VN otherwise, so cleaning up.

### CODE REVIEW

File: testsuite/testcases/src/modifiers.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                        name,
+                        e
+                    )
+                })?;
+        }
+        Ok::<(), anyhow::Error>(())
+    })
+}
+
+/// Config for additing variable processing overhead/delay into
```
Comment (JoshLind) [inline testsuite/testcases/src/modifiers.rs:74]: `additing` -> `adding`?

**Review by JoshLind — APPROVED** [review]
Looks reasonable to me 😄

**Review by sitalkedia — APPROVED** [review]
LGTM

File: testsuite/testcases/src/modifiers.rs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    runtime.block_on(async {
+        let mut rng = rand::thread_rng();
+        for (name, validator) in validators {
+            let sleep_percentage = if rng.gen_bool(config.inject_delay_node_fraction) {
+                rng.gen_range(1_u32, config.inject_delay_max_transaction_percentage)
+            } else {
+                0
+            };
+            let name = name.clone();
```
[the remark this answers is not in the document]
Comment (igor-aptos) [reply testsuite/testcases/src/modifiers.rs]: hm, it was needed a while back for some reason, now doesn't seem like it is needed anymore

**Pull request merged**